### DISCUSSION TITLE
Some questions regarding LED Status Indicators

### DISCUSSION THREAD

**Discussion by Valour549** [seerge/g-helper#3108] [Q&A] [answered]
I made myself familiar with the [Readme](https://github.com/seerge/g-helper?tab=readme-ov-file), [FAQ](https://github.com/seerge/g-helper/wiki/FAQ) and [Troubleshooting](https://github.com/seerge/g-helper/wiki/Troubleshooting).

I have the Asus Scar 18 (2024) and G-helper works wonderfully in toggling the Status indicator LEDs. I don't see any options in BIOS or Armory Crate (correct me if I'm just blind), but clearly the option is there somewhere!

1) Could anyone share how this was done? I would like a script (or two scripts) that I can use to automate toggling on/off the LEDs when the laptop turns on/off in combination with Task Scheduler. You see I would like them to be on when the laptop is on, and to be off when the laptop turns off. This is because the laptop is plugged in all the time, and when it is off the battery LED indicator is still white, which annoys me when I sleep lol.

2) I also noticed that the toggle affects [all four indicators](https://i.imgur.com/0IYJ8OR.jpeg) including the power button, is it possible to leave the power button on and only toggle the other three indicators?

3) A slightly unrelated question: For the keyboard/logo/lightbar/lid backlight, if "Shutdown" is ticked, does that mean they will all be on even when the laptop is shut down? Because I have it ticked and they all turn off when the laptop turns off. I have also unticked them and can't observe any difference.

Thanks!

**Comment by seerge** [comment]
@Valour549 hello

1. This is done via ON / OFF toggle endpoint exposed by Asus for AC (initially)
https://github.com/seerge/g-helper/blob/ef769982e0a142afd80059dd3a55d75ec9976e9e/app/AsusACPI.cs#L133

2. You can't goggle LEDs individually. Toggle is very simple and affects all them together

3. "Shutdown" (same as "Startup", "Sleep" or "Awake") stand for separate backlight "states" keyboard and other zones could be. Shutdown is usually a double-blink before laptop completely powers off. You can toggle it off if you don't like it.

**Reply by seerge** [reply, marked as answer]
You can try to run following powershell command as admin

```
Invoke-CimMethod (Get-CimInstance -Namespace root/wmi -ClassName AsusAtkWmi_WMNB) -MethodName DEVS -Arguments @{Device_ID=0x000600C2; Control_status=0}
````

0 / 1 is the status. But i'm not sure if that will work in your case, as not all endpoints are accessible via WMI commands :)

**Reply by Valour549** [reply]
1. How would I make a .bat file that would toggle on/off the StatusLED? Could you possibly provide one?
2. Got it.
3. I see, I'll observe the patterns more carefully.
 
Thank you for the prompt reply.

**Reply by seerge** [reply]
@Valour549 I just gave you powershell command. You can try that if it works or not. If it doesn't you **can't** do it via bat (you will need to write c# or c++ or whatever code to access same endpoint via driver)

**Reply by Valour549** [reply]
@seerge Works perfectly! I made them into exes with cute little icons if anyone ever wants to use them lol.

https://github.com/Valour549/Scar18-LED-Toggle

**Reply by Valour549** [reply]
@seerge Is it possible to modify your powershell command so it becomes a toggle? Something like: If Control_status=0 make it 1 and vice versa.

**Reply by seerge** [reply]
@Valour549 hello, it's tricky cause you can't "read" current status from BIOS most likely. Only set it. 

So you need to store last set status locally, and then based on that do a toggle of course :)

**Reply by Valour549** [reply]
@seerge You're right, I found the LED toggle option in Armory Crate settings (it was there after all, I just didn't look hard enough at the start), and it was unable to read the correct status when I changed it using the powershell commands.

Would you be able to write a powershell script for me so it can toggle using the local status thing? My Scar 18 only comes with 5 customizable hotkeys so it'll be good to only bind one of those to toggle LED status instead of two lol
